src/sql/src/plan/statement/ddl/source.rs

```diff
@@ -85,663 +85,981 @@ pub fn describe_create_subsource(
     Ok(StatementDesc::new(None))
 }
 
-generate_extracted_config!(
-    CreateSourceOption,
-    (IgnoreKeys, bool),
-    (Timeline, String),
-    (TimestampInterval, Duration),
-    (RetainHistory, Duration)
-);
-
 generate_extracted_config!(
     PgConfigOption,
     (Details, String),
     (Publication, String),
     (TextColumns, Vec::<UnresolvedItemName>, Default(vec![]))
 );
 
-generate_extracted_config!(
-    MySqlConfigOption,
-    (Details, String),
-    (TextColumns, Vec::<UnresolvedItemName>, Default(vec![])),
-    (IgnoreColumns, Vec::<UnresolvedItemName>, Default(vec![]))
-);
-
-pub fn plan_create_source(
+fn plan_create_source_desc_postgres(
     scx: &StatementContext,
-    mut stmt: CreateSourceStatement<Aug>,
-) -> Result<Plan, PlanError> {
+    stmt: &CreateSourceStatement<Aug>,
+) -> Result<(SourceDesc<ReferencedConnection>, RelationDesc), PlanError> {
     mz_ore::soft_assert_or_log!(
         stmt.referenced_subsources.is_none(),
         "referenced subsources must be cleared in purification"
     );
 
     let CreateSourceStatement {
-        name,
+        name: _,
         in_cluster: _,
-        col_names,
+        col_names: _,
         connection,
         envelope,
-        if_not_exists,
+        if_not_exists: _,
         format,
-        key_constraint,
+        key_constraint: _,
         include_metadata,
         with_options,
         referenced_subsources: _,
-        progress_subsource,
+        progress_subsource: _,
     } = &stmt;
 
-    let envelope = envelope.clone().unwrap_or(ast::SourceEnvelope::None);
+    let CreateSourceOptionExtracted {
+        timeline: _,
+        timestamp_interval,
+        ignore_keys: _,
+        retain_history: _,
+        seen: _,
+    } = CreateSourceOptionExtracted::try_from(with_options.clone())?;
 
-    let allowed_with_options = vec![
-        CreateSourceOptionName::TimestampInterval,
-        CreateSourceOptionName::RetainHistory,
-    ];
-    if let Some(op) = with_options
-        .iter()
-        .find(|op| !allowed_with_options.contains(&op.name))
-    {
-        scx.require_feature_flag_w_dynamic_desc(
-            &vars::ENABLE_CREATE_SOURCE_DENYLIST_WITH_OPTIONS,
-            format!("CREATE SOURCE...WITH ({}..)", op.name.to_ast_string()),
-            format!(
-                "permitted options are {}",
-                comma_separated(&allowed_with_options)
-            ),
-        )?;
+    for (check, feature) in [
+        (
+            matches!(envelope, Some(ast::SourceEnvelope::None) | None),
```

Comment:
do we even want to accept an _explicit_ `ENVELOPE NONE`? Not worth changing now, but we probably want to come back to this

Reply:
yeah this was a consideration but I wanted to see if I could avoid writing a migration just in case someone had done this.